web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java

```diff
@@ -64,48 +65,39 @@ class FilterOnSickLeaveLength {
 
         @Test
         void shouldFilterOnSickLeaveLengthIfSpecified() {
-            final var expectedSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 12, PATIENT_ID);
+            final var expectedSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 36, PATIENT_ID);
             final var expectedSickLeaveList = List.of(expectedSickLeave);
-            final var anotherSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 5, PATIENT_ID);
-            final var sickLeaves = List.of(expectedSickLeave, anotherSickLeave);
-
-            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, 10, 15, Collections.emptyList(), null, null);
-
-            assertEquals(expectedSickLeaveList, actualSickLeaveList);
-        }
-
-        @Test
-        void shouldNotFilterOnSickLeaveLengthIfFromAndToIsNull() {
-            final var expectedSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 12, PATIENT_ID);
-            final var anotherSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 5, PATIENT_ID);
-            final var expectedSickLeaveList = List.of(expectedSickLeave, anotherSickLeave);
+            final var anotherSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 42, PATIENT_ID);
             final var sickLeaves = List.of(expectedSickLeave, anotherSickLeave);
 
-            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, null, Collections.emptyList(), null, null);
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves,
+                List.of(new SickLeaveLengthInterval(6, 12), new SickLeaveLengthInterval(12, 40), new SickLeaveLengthInterval(50, 100)),
```

Comment:
We should add tests to see if filtering is correct when there is either from or to as null

Reply:
My bad, i was not aware of that from/to could be null. Updated!